File: magellan-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.java
package com.wealthfront.magellan.sample;

import android.os.Bundle;

import com.wealthfront.magellan.Navigator;
import com.wealthfront.magellan.support.SingleActivity;

public class MainActivity extends SingleActivity {

    @Override
    protected Navigator createNavigator() {
        return Navigator.withRoot(new HomeScreen()).loggingEnabled(true).build();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
    }

}

File: magellan-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt
package com.wealthfront.magellan.sample

import android.os.Bundle
import androidx.activity.ComponentActivity
import com.example.magellan.compose.setContentNavigable
import com.wealthfront.magellan.sample.App.Provider.appComponent
import javax.inject.Inject

class MainActivity : ComponentActivity() {

  @Inject lateinit var expedition: Expedition

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    appComponent.inject(this)
    setContentNavigable(expedition)
  }

  override fun onBackPressed() {
    if (!expedition.backPressed()) {
      super.onBackPressed()
    }
  }
}
